src/coreclr/src/jit/hwintrinsic.cpp

```diff
@@ -643,6 +643,11 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
     baseType          = getBaseTypeFromArgIfNeeded(intrinsic, clsHnd, sig, baseType);
     unsigned simdSize = HWIntrinsicInfo::lookupSimdSize(this, intrinsic, sig);
 
+    if (!varTypeIsArithmetic(baseType))
```

Comment:
@tannergooding - I see that you recommended this. Can you explain why this wasn't needed previously?

Reply:
The majority of intrinsics aren't generic and we were treating the couple that were (`Vector128<T>.Count` and `Vector128<T>.Zero`) as `SpecialImport`.

This allows them to be table driven instead as it handles the case where a user tries `Vector128<bool>.Zero`, for example.

Reply:
Great - thanks for clarifying

Reply:
This change is no longer existent, as we marked this as `SpecialImport` again.